### PR TITLE
Homepage Hero Animation

### LINKED ISSUE #52: Homepage Hero
## Description
Once we have added all the styles and the common elements to the web, we'll still need to adapt the animation for the terminal and the video created by the motion designer in the homepage hero.

Screenshot from 2020-03-20 13-12-47

Note that the different `clip-path`s don't apply in this issue, they are going to be managed in another one.

## Logic
We'll have two different elements in the same group: the `terminal` (that gets its style from this [processor](https://github.com/frontity/frontity.org/issues/20)) and the video made from the motion designer, the one in the back. Some things we have to achieve:

* We need to modify slightly the `terminal` element, and the lines should be filled letter by letter just like [Strapi](https://strapi.io/) is doing. We could use a library like [React Typist](https://github.com/jstejada/react-typist). (We still have to define the time between letters and lines with the designer).
* When the commands have finished, the terminal has to go down an hide to show the video in the background.
* There will be a button to show/hide the terminal.

It should look similar to this test, but adapting the styles to the Invision design -> https://www.loom.com/share/3ab07ac6d46749d6a897f575c357eae5

## Possible solution
I've linked a draft PR with the code I used to implement the test shown in the previous video, just in case you find it useful. #76 

Bear in mind that this solution is hardcoded. We should try to reuse the `terminal` and `image-frame` processors and, if possible, get the content from WordPress (if you feel it'd too complicated, there's no problem hardcoding this content).

### PR DESCRIPTION
**Pull request by iamuchejude**
Fixed #52

### CODE REVIEW

**Comment by frontibotito** [comment]
Deploy preview for _website_ ready!

Built with commit d681935d159ad132b8cd17108fcba80a0c132cfe

https://frontity-org-k449rnp2l.now.sh

File: packages/frontity-org-theme/src/processors/homepage-hero.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { Processor } from "@frontity/html2react/types";
+import { HTMLProps } from "react";
+
+import HomepageHero from "../components/homepage-hero";
+
+export const homepageHeroAnimation: Processor<HTMLProps<Element>> = {
+  name: "homepage-hero-animation",
+  test: ({ node }) =>
+    node.type === "element" &&
+    node.props.className.split(" ").includes("homepage-hero-animation"),
```
Comment (DAreRodz) [inline packages/frontity-org-theme/src/processors/homepage-hero.ts:11]: Hey, you forgot to add a `?` after `className`. If you look into the console logs you will see a lot of warning messages saying `Cannot read "split" of undefined`.

**Comment by SantosGuillamot** [comment]
@iamuchejude As we talked, let's load a video instead of an image (keeping the image as a fallback) once the terminal finish loading. This is the video we're going to use:
https://wp.frontity.org/wp-content/uploads/2020/04/Setting-Up-Frontity.mp4

File: packages/frontity-org-theme/src/processors/homepage-hero.ts (hunk 1 above)
Comment (iamuchejude) [reply packages/frontity-org-theme/src/processors/homepage-hero.ts:11]: Oh! Thanks, David.

**Comment by SantosGuillamot** [comment]
@DAreRodz I've pushed the fix of the console warnings and a few styles. Could you review it?

**Comment by iamuchejude** [comment]
@SantosGuillamot I think we should work with the image (instead of video) for the main time, pending when the video is completely implemented. I stumbled upon a couple of issues that have delayed the task to replace the image with the video.

Also, I'm gonna fix the merge conflict on the PR so it could be merged as soon as possible.

**Pull request merged**